### DISCUSSION TITLE
Insert UI element into existing component

### DISCUSSION THREAD

**Discussion by GuillaumeGomez** [bevyengine/bevy#6521] [Q&A] [answered]
I'm working on a small [video game](https://github.com/GuillaumeGomez/rust-toy-game/tree/bevy) and I encountered the following issue: I cannot find a way to insert a child into an existing element. 

I get the `Entity` from a query:

```rust
characters: &mut Query<(Entity, &mut Character, &Children)>,
```

and then use it like this (entity is `receiver_id`):

```rust
commands.entity(receiver_id) // I suppose the `receiver_id` is wrong here
    .add_children(|children| {
        children.spawn_bundle(TextBundle::from_section(
            damage.to_string().as_str(),
            TextStyle {
                font: asset_server.load("fonts/kreon-regular.ttf"),
                font_size: 10.0 / crate::SCALE,
                color: Color::RED,
            },
        ))
        .insert(Notification {
            timer: Timer::from_seconds(NOTIFICATION_TIME, false),
        })
        .insert_bundle(TransformBundle::from(Transform::from_xyz(
            0.,
            receiver.height / 2. + 2.,
            0.,
        )));
    });
```

I suppose I'm doing something wrong because then it's not displayed and I get a lot of warnings:

```
2022-11-08T23:34:45.153193Z  WARN bevy_ui::flex: Styled child in a non-UI entity hierarchy. You are using an entity with UI components as a child of an entity without UI components, results may be unexpected.
```

However I checked and `receiver_id` is the same ID as the one returned when I created the `Character` with `commands.spawn()`.

Any idea how I can fix this issue?

**Comment by hymm** [comment, marked as answer]
TextBundle can only be used in ui. You probably want a Text2dBundle instead https://docs.rs/bevy/latest/bevy/text/struct.Text2dBundle.html

**Reply by GuillaumeGomez** [reply]
Thanks a lot! It was exactly the problem. Working exactly as expected now.
